jerry-core/ecma/base/ecma-helpers-string.c

```diff
@@ -2458,17 +2457,30 @@ ecma_string_trim_helper (const lit_utf8_byte_t **utf8_str_p, /**< [in, out] curr
   while (current_p > nonws_start_p)
   {
     read_size = lit_read_prev_code_unit_from_utf8 (current_p, &ch);
-
     if (!lit_char_is_white_space (ch))
     {
       break;
     }
-
     current_p -= read_size;
   }
 
-  *utf8_str_p = nonws_start_p;
-  *utf8_str_size = (lit_utf8_size_t) (current_p - nonws_start_p);
+  if (trim_position == ECMA_TRIM_STRING_END)
+  {
+    *utf8_str_size = *utf8_str_size - (lit_utf8_size_t) ((*utf8_str_p + *utf8_str_size) - current_p);
+  }
+  else
+  {
+    if (trim_position == ECMA_TRIM_STRING_START)
+    {
+      *utf8_str_size = (lit_utf8_size_t) ((*utf8_str_p + *utf8_str_size) - nonws_start_p);
+    }
+    else
+    {
+      *utf8_str_size = (lit_utf8_size_t) (current_p - nonws_start_p);
+    }
+
+    *utf8_str_p = nonws_start_p;
+  }
```

Comment:
I think this should be performed in an other way. Currently you trim the string from both directions after you calculate back the required string part which is superfluous.

```c
if (flags & TRIM_START) {
  //  remove leading whitespaces
}

if (flags & TRIM_END) {
  //  remove trailing whitespaces
}

// return as before
*utf8_str_p = nonws_start_p;
*utf8_str_size = (lit_utf8_size_t) (current_p - nonws_start_p);
```

Reply:
My problem with this atm, is that if i don't set the `nonws_start_p` in the forward trimming part, than the second part, which checks for `current_p > nonws_start_p` will never pass, as `nonws_start_p` and `current_p` will both be set to `*utf8_str_p + *utf8_str_size`. This will lead to the `utf8_str_p` to be set to the end of the string on trimend functions.

This is the reason why i do this rather ugly code atm. Although i could set `nonws_start_p` to the original `utf8_str_p` when i'm doing trimEnd, which will lead to the if passing, and as i don't really care about what is on the front of the string, this might be a solution for this. Any thoughts?

Reply:
I think this can be resolved with 3 variables. start, current, end

then:

```
current = start;

while (current < end) {
  //trim
}

start = current;
current = end;

while (current < end) {
 // trim
}
*utf8_str_p = start_p;
*utf8_str_size = (lit_utf8_size_t) (end_p - start_p);
```

Reply:
Why don't we split the trim helper into `ecma_string_trim_front`/`ecma_string_trim_back`?

Reply:
With a nicer solution than mine (such as @rerobika's idea) it would be unnecessary to cut this, even more so if only these functions use the front/end trimming part imo.

Reply:
The question isn't whether it is necessary or not, it's whether we should. We don't gain anything by keeping the trimming code in a single function, even if separated nothing would be duplicate code. In it's current form all we do is add superfluous conditions, which wouldn't be necessary if they are split properly.

Reply:
Okay, i'll split this into corresponding sub-functions.